atomate/vasp/firetasks/surface_tasks.py

```diff
@@ -0,0 +1,174 @@
+                    else:
+                        # build a oriented unit cell fw
+                        FWs.append(self.get_ouc_fw(slab))
+                        miller_list.append(tuple(slab.miller_index))
+
+        elif self.get("structure_type") == "oriented_unit_cell":
+
+            folder = os.path.basename(os.getcwd())
+            # If this is a reconstruction, we need to use the ReconstructionGenerator
+            if "_rec_" in folder:
+                ouc = Structure.from_file("CONTCAR.relax2.gz")
+                # ReconstructionGenerator only works on the conventional ucell
+                ucell = SpacegroupAnalyzer(ouc).get_conventional_standard_structure()
+                # Get the name of the reocnstruction
+                ns, n = "", 0
+                for s in folder:
+                    ns+=s
+                    if s == "_":
+                        n+=1
+                    if n == 5:
+                        break
+                rec = ReconstructionGenerator(ucell, slab_gen_params["min_slab_size"],
+                                              slab_gen_params["min_vacuum_size"],
+                                              reconstruction_name=folder[len(ns):])
+                FWs.append(self.get_slab_fw(rec.build_slab(), slab_gen_params))
+
+            else:
+                # Get the list of FWs for the various terminations
+                # for a slab based on the oriented unit cell
+                slab_gen_params["initial_structure"] = \
+                    Structure.from_file("CONTCAR.relax2.gz")
+                slab_gen_params["miller_index"] = [0,0,1]
+                symmetrize = slab_gen_params["symmetrize"]
+                del slab_gen_params["symmetrize"]
+                slabgen = SlabGenerator(**slab_gen_params)
+                for slab in slabgen.get_slabs(symmetrize=symmetrize):
+                    slab.miller_index = self.get("miller_index")
+                    FWs.append(self.get_slab_fw(slab, slab_gen_params))
+
+        return FWAction(additions=FWs)
+
+    def get_ouc_fw(self, slab):
+        """
+        Return a SurfCalcOptimizer FW for an oriented unit cell.
+
+        Args:
+            slab (Slab): Slab object containing various
+                attributes related to the slab
+        """
+
+        from atomate.vasp.fireworks.core import SurfCalcOptimizer
+        return SurfCalcOptimizer(slab.oriented_unit_cell, self.get("scratch_dir"),
+                                 self.get("k_product"), self.get("vasp_cmd"),
+                                 "oriented_unit_cell", self.get("cwd", os.getcwd()),
+                                 reconstruction=slab.reconstruction,
+                                 miller_index=slab.miller_index,
+                                 db_file=self.get("db_file"),
+                                 scale_factor=slab.scale_factor,
+                                 mpid=self.get("mpid", "--"))
+
+    def get_slab_fw(self, slab, slab_gen_params):
```

Comment:
Some of this slab generation etc. seems to be common with the existing adsorption workflows. Can you check with @montoyjh if there is some way to keep consistency and a common set of tools, e.g. for slab generation, between the two sets of workflows?